### PR TITLE
Devise User System and Google sign in

### PR DESCRIPTION
**Pull request by bdonald25**
#2370 New user system and support for google sign in.

### CODE REVIEW

**Comment by bdonald25** [comment]
@pat-son @tylerpuleo Can you both review this since it is so big? 

Also, circle stopped working because one of the dependencies started prompting "Are you sure Y|n
" so I fixed it by piping 'yes' into the command, so this needs to get merged for the other builds to start passing again.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -55,7 +93,7 @@ def create_issue_anon
   end
 
   def find_user
-    @cur_user = User.find_by_id(session[:user_id])
+    #     current_user = User.find_by_id(session[:user_id])
```
Comment (pat-son) [inline app/controllers/application_controller.rb]: Does this need to stay here commented out?

File: config/locales/devise.en.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    confirmations:
+      confirmed: "Your email address has been successfully confirmed."
+      send_instructions: "You will receive an email with instructions for how to confirm your email address in a few minutes."
+      send_paranoid_instructions: "If your email address exists in our database, you will receive an email with instructions for how to confirm your email address in a few minutes."
+    failure:
+      already_authenticated: "You are already signed in."
+      inactive: "Your account is not activated yet."
+      invalid: "Invalid %{authentication_keys} or password."
+      locked: "Your account is locked."
+      last_attempt: "You have one more attempt before your account is locked."
```
Comment (tylerpuleo) [inline config/locales/devise.en.yml]: attempts

File: config/locales/devise.en.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      not_found_in_database: "Invalid %{authentication_keys} or password."
+      timeout: "Your session expired. Please sign in again to continue."
```
Comment (tylerpuleo) [inline config/locales/devise.en.yml]: has expired

File: config/locales/devise.en.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      failure: "Could not authenticate you from %{kind} because \"%{reason}\"."
+      success: "Successfully authenticated from %{kind} account."
+    passwords:
+      no_token: "You can't access this page without coming from a password reset email. If you do come from a password reset email, please make sure you used the full URL provided."
+      send_instructions: "You will receive an email with instructions on how to reset your password in a few minutes."
+      send_paranoid_instructions: "If your email address exists in our database, you will receive a password recovery link at your email address in a few minutes."
+      updated: "Your password has been changed successfully. You are now signed in."
+      updated_not_active: "Your password has been changed successfully."
+    registrations:
+      destroyed: "Bye! Your account has been successfully cancelled. We hope to see you again soon."
```
Comment (tylerpuleo) [inline config/locales/devise.en.yml]: something meaner than this :)

File: db/schema.rb [hunk 5]
```diff
@@ -118,19 +118,31 @@
     t.string   "lastname"
     t.string   "email"
     t.integer  "group_id"
-    t.boolean  "validated",       default: false
+    t.boolean  "validated",              default: false
```
Comment (tylerpuleo) [inline db/schema.rb]: Is that an extra tab?

File: db/schema.rb [hunk 6]
```diff
@@ -118,19 +118,31 @@
...
     t.string   "password_digest"
     t.datetime "created_at"
     t.datetime "updated_at"
     t.string   "validation_key"
-    t.boolean  "admin",           default: false
-    t.boolean  "hidden",          default: false
+    t.boolean  "admin",                  default: false
```
Comment (tylerpuleo) [inline db/schema.rb]: and this

File: db/schema.rb [hunk 7]
```diff
@@ -118,19 +118,31 @@
...
+    t.boolean  "hidden",                 default: false
```
Comment (tylerpuleo) [inline db/schema.rb]: and this

File: db/schema.rb [hunk 8]
```diff
@@ -118,19 +118,31 @@
...
     t.text     "bio"
     t.integer  "news_id"
-    t.datetime "last_login",      default: '2013-08-16 12:00:00'
+    t.datetime "last_login",             default: '2013-08-16 12:00:00'
```
Comment (tylerpuleo) [inline db/schema.rb]: and this

File: public/assets/btn_google_light_normal_ios.svg [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+                <path d="M3.96409091,10.71 C3.78409091,10.17 3.68181818,9.59318182 3.68181818,9 C3.68181818,8.40681818 3.78409091,7.83 3.96409091,7.29 L3.96409091,4.95818182 L0.957272727,4.95818182 C0.347727273,6.17318182 0,7.54772727 0,9 C0,10.4522727 0.347727273,11.8268182 0.957272727,13.0418182 L3.96409091,10.71 L3.96409091,10.71 Z" id="Shape" fill="#FBBC05" sketch:type="MSShapeGroup"></path>
+                <path d="M9,3.57954545 C10.3213636,3.57954545 11.5077273,4.03363636 12.4404545,4.92545455 L15.0218182,2.34409091 C13.4631818,0.891818182 11.4259091,0 9,0 C5.48181818,0 2.43818182,2.01681818 0.957272727,4.95818182 L3.96409091,7.29 C4.67181818,5.16272727 6.65590909,3.57954545 9,3.57954545 L9,3.57954545 Z" id="Shape" fill="#EA4335" sketch:type="MSShapeGroup"></path>
+                <path d="M0,0 L18,0 L18,18 L0,18 L0,0 Z" id="Shape" sketch:type="MSShapeGroup"></path>
+            </g>
+            <g id="handles_square" sketch:type="MSLayerGroup"></g>
+        </g>
+    </g>
+</svg>
```
Comment (tylerpuleo) [inline public/assets/btn_google_light_normal_ios.svg]: newline

File: test/functional/users_controller_test.rb [hunk 10]
```diff
@@ -2,83 +2,65 @@
...
-    end
-
-    assert_redirected_to user_path(assigns(:user))
-
-    john = User.find_by_email('john@example.com')
-
-    assert_not_nil john
-    assert_equal john.validated?, false,
-      'New user should not be validated'
+    skip('No longer valid as of devise integration')
```
Comment (tylerpuleo) [inline test/functional/users_controller_test.rb]: This is kind of weird. Shouldn't we just take out the test

File: test/test_helper.rb [hunk 11]
```diff
@@ -143,3 +144,7 @@ def wait_for_ajax
     end
   end
 end
+
+class ActionController::TestCase
+  include Devise::TestHelpers
+end
```
Comment (tylerpuleo) [inline test/test_helper.rb]: newline

File: test/integration/log_in_test.rb [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+  test 'should not log in because of bad password' do
+    visit '/'
+    click_on('Login')
+    fill_in 'user_email', with: 'nixon@whitehouse.gov'
+    fill_in 'user_password', with: '123456'
+    find(:css, '.mainContent').click_on('Log in')
+
+    assert page.has_content?('Invalid email or password.'), 'Did not fail to log in'
+  end
+end
```
Comment (tylerpuleo) [inline test/integration/log_in_test.rb]: newline

File: db/schema.rb (hunk 5 above)
Comment (bdonald25) [inline db/schema.rb]: schema is auto generated to reflect migrations

File: public/assets/btn_google_light_normal_ios.svg (hunk 9 above)
Comment (bdonald25) [inline public/assets/btn_google_light_normal_ios.svg]: This is google's logo they require for google log in, I just copied the file that they require into our public assets. Take it up with google.

File: config/locales/devise.en.yml (hunk 2 above)
Comment (bdonald25) [inline config/locales/devise.en.yml]: "one more attempts" I don't think so.

File: config/locales/devise.en.yml (hunk 4 above)
Comment (bdonald25) [inline config/locales/devise.en.yml]: "Your account has been successfully cancelled, your loss!"

File: test/functional/users_controller_test.rb (hunk 10 above)
Comment (bdonald25) [inline test/functional/users_controller_test.rb]: Yes, I just left it like that so we know what tests are being removed and no one thinks we are just removing a ton of tests. I am planning on removing all the skipped tests in another pull request so it is clear what ones are being removed and why.

**Pull request merged**